### ISSUE TITLE
clear_database does not reset PK counter in the test DB

### ISSUE THREAD

**Issue by Tseplyaev** [aiidateam/aiida-core#3721]
Writing tests, I faced an inconvenience: PKs of created test nodes 
are not constant in a given test despite clearing the DB via `clear_database` fixture.

Let me show an example:
let us say I have defined a new fixture which is called 'clear_db_before' (which duplicates `clear_database`):
```
@pytest.fixture(scope='function', autouse=True)
def clear_database_before(aiida_profile):
    """Clear the database before each test.
    """
    aiida_profile.reset_db()
```
and a test:
```
def test_something():
    from aiida.orm import Dict
    test_dict = Dict(dict={})
    test_dict.store()
    assert test_dict.pk == 1
```

The test passes when I submit only this one is a session, however it fails when I submit all the tests I have in a single run. It happens because the `Dict` inside `test_something` does not get the same PK as it had before - now it gets a higher value of PK and `assert test_dict.pk == 1` fails. 

For instance, let us say the `test_dict` gets PK=67 ⟹ I assume there should be a node with PK=42. However, I can not load it like:
```
def test_something():
    from aiida.orm import Dict, load_node
    test_dict = Dict(dict={})
    test_dict.store()
    random_node = load_node(42) # fails with NotExistent
    assert test_dict.pk == 1  # test_dict.pk == 67 when all tests submitted
```
 which proves that the DB was cleared but for some reason PK counter was not reset.

To summarise the issue I have a question:
why does not `clear_database` reset a PK counter but removes all the nodes?

**Comment by sphuber** [comment]
This is because in the `clear_database` the tables are merely emptied. The tables itself including the sequence counters are kept in tact, as you have noticed. Typically, you do not want to depend on explicit pks exactly for this reason. Instead probably you want to query directly for something you just created, or otherwise you can `QueryBuilder().append(Dict).count() == 1` to assure there is only one node present. Is there any specific test where you really _have_ to know the pk?

**Comment by Tseplyaev** [comment]
Yes, I want to test a function that analyses the content of my DB and I want to make sure that it writes correct node list and their PKs to the output .json file. However, I did not really have a problem - it was solved by saving actual node PKs before executing analysing function.

Thank you for clarifying, I guess I will always save node PKs upon storing them and do not make hardcoded checks as in an example above. I did not know the details about clearing the DB and assumed that it is one of the reasons to clear the database between two tests in a session - to ensure constant PKs.

**Comment by sphuber** [comment]
> I guess I will always save node PKs upon storing them and do not make hardcoded checks as in an example above

Yes this is exactly the advised way if you really need to compare explicit pks. If this has answered your question, can we close this issue?

**Comment by Tseplyaev** [comment]
Yes, Sebastiaan, you answered my question. Thank you!
